x/auth/vesting/msg_server.go

```diff
@@ -256,5 +185,173 @@ func (s msgServer) CreatePeriodicVestingAccount(goCtx context.Context, msg *type
 		),
 	)
 	return &types.MsgCreatePeriodicVestingAccountResponse{}, nil
+}
+
+func max64(a, b int64) int64 {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+// CreateTrueVestingAccount creates a new "true" vesting account, or merges a grant into an existing one.
+func (s msgServer) CreateTrueVestingAccount(goCtx context.Context, msg *types.MsgCreateTrueVestingAccount) (*types.MsgCreateTrueVestingAccountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	ak := s.AccountKeeper
+	bk := s.BankKeeper
+
+	from, err := sdk.AccAddressFromBech32(msg.FromAddress)
+	if err != nil {
+		return nil, err
+	}
+	to, err := sdk.AccAddressFromBech32(msg.ToAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	if bk.BlockedAddr(to) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, "%s is not allowed to receive funds", msg.ToAddress)
+	}
+
+	vestingCoins := sdk.NewCoins()
+	for _, period := range msg.VestingPeriods {
+		vestingCoins = vestingCoins.Add(period.Amount...)
+	}
+
+	lockupCoins := sdk.NewCoins()
+	for _, period := range msg.LockupPeriods {
+		lockupCoins = lockupCoins.Add(period.Amount...)
+	}
+
+	if !vestingCoins.IsZero() && len(msg.LockupPeriods) == 0 {
+		// If lockup absent, default to an instant unlock schedule
+		msg.LockupPeriods = []types.Period{
+			{Length: 0, Amount: vestingCoins},
+		}
+		lockupCoins = vestingCoins
+	}
+
+	if !lockupCoins.IsZero() && len(msg.VestingPeriods) == 0 {
+		// If vesting absent, default to an instant vesting schedule
+		msg.VestingPeriods = []types.Period{
+			{Length: 0, Amount: lockupCoins},
+		}
+		vestingCoins = lockupCoins
+	}
+
+	if !vestingCoins.IsEqual(lockupCoins) { // XXX IsEqual can crash
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "lockup and vesting amounts must be equal")
+	}
+
+	madeNewAcc := false
+	acc := ak.GetAccount(ctx, to)
+
+	if acc != nil {
+		pva, ok := acc.(*types.TrueVestingAccount)
```

Comment:
Please rename `pva` to `va` here and below, so that we can pivot on the account names without leaving behind cruft from older names:
```suggestion
		va, ok := acc.(*types.TrueVestingAccount)
```

Reply:
Done.